helper.go

```diff
@@ -442,62 +439,59 @@ func marshalLength(length int) ([]byte, error) {
 	return append(header, bufBytes...), nil
 }
 
-func marshalObjectIdentifier(oid []int) (ret []byte, err error) {
+func marshalObjectIdentifier(oid string) ([]byte, error) {
 	out := new(bytes.Buffer)
-	if len(oid) < 2 || oid[0] > 6 || oid[1] >= 40 {
-		return nil, errors.New("invalid object identifier")
-	}
-
-	err = out.WriteByte(byte(oid[0]*40 + oid[1]))
-	if err != nil {
-		return
-	}
-	for i := 2; i < len(oid); i++ {
-		err = marshalBase128Int(out, int64(oid[i]))
-		if err != nil {
-			return
-		}
-	}
-
-	ret = out.Bytes()
-	return
-}
-
-func marshalOID(oid string) ([]byte, error) {
+	oidLength := len(oid)
+	oidBase := 0
 	var err error
+	i := 0
+	for j := 0; j < oidLength; {
+		if oid[j] == '.' {
+			j++
+			continue
+		}
+		var val int64 = 0
+		for j < oidLength && oid[j] != '.' {
+			ch := int64(oid[j] - '0')
+			if ch > 9 {
```

Comment:
Or `< 0` ?

Reply:
It's not necessary. The original byte at the index position (oid[j]) is of type uint8. The expression oid [j] - '0' will also always be > 0.